src/pages/workspaces/workspace/jobHistory/CallTable.js

```diff
@@ -346,6 +347,37 @@ const CallTable = ({
             },
             ...(isAzure // GCP workspaces do not have inputs/outputs/logs (only log file is at top level, linked in workflow dashboard)
               ? [
+                  {
+                    size: { basis: 200, grow: 1 },
+                    field: 'cost',
+                    headerRenderer: () =>
+                      h(Sortable, { sort, field: 'cost', onSort: setSort }, [
+                        'Approximate Cost',
+                        h(
+                          TooltipTrigger,
+                          {
+                            content:
+                              'Approximate cost is calculated based on the list price of the VM, and does not include disk cost or any cloud account discounts.',
+                          },
+                          [icon('info-circle', { style: { marginLeft: '0.4rem', color: colors.accent(1) } })]
+                        ),
+                      ]),
+                    cellRenderer: ({ rowIndex }) => {
+                      const { vmCostUsd, taskStartTime, taskEndTime, executionStatus, callCaching, subWorkflowId } = filteredCallObjects[rowIndex];
+                      if (taskStartTime && vmCostUsd) {
+                        if (executionStatus === 'Running') {
```

Comment:
There's a lot of if-nesting here, and I have questions like "but what about state X" (eg "aborting"). I also had a similar question to Janet below. I wonder, would it be easier to use `if (endTime)` here instead of `if (executionStatus == 'Running')`? And then have something like - 

```js
// taskHasACost could be a testable function, equivalent to 'tastStartTime && vmCostUsd'
if (taskHasACost) {
  if (taskEndTime) {
    // Calculate with end time
  } else { 
    // Calculate without end time
  }
} else {
  // noCostExpected could be a separate function with its own unit tests and clear semantics 
  if (noCostExpected) {
    return '$0.00'
  } else {
    return 'Cost pending'
  }
}
```